localstack-core/localstack/services/stepfunctions/provider.py

```diff
@@ -508,7 +540,17 @@ def start_execution(
             normalised_state_machine_arn, exec_name
         )
         if exec_arn in self.get_store(context).executions:
-            raise InvalidName()  # TODO
+            # Return already running execution if name and input match
+            existing_execution = self._idempotent_start_execution(
+                context=context,
+                exec_arn=exec_arn,
+                state_machine=state_machine_clone,
+                name=name,
+                input_data=input_data,
+            )
+
+            if existing_execution:
```

Comment:
I see now, this function returns None after sampling the store for the execution arn! A similar check is run by the provider before invoking `_idempotent_start_execution`. Maybe we can access the store only once and revise the flow slightly here to avoid confusions

Reply:
I added the store check to the `_idempotent_start_execution` function to make it better suited to running in isolation. Agree that the double check is unneccesary so happy to remove it.